packages/data-point/lib/reducer-types/resolve.js

```diff
@@ -15,28 +16,55 @@ const reducers = Object.assign({}, ReducerHelpers, {
   [ReducerPath.type]: ReducerPath
 })
 
+/**
+ * @param {Reducer} reducer
+ * @return {boolean}
+ */
+function hasDefault (reducer) {
+  return !!reducer[DEFAULT_VALUE]
+}
+
+module.exports.hasDefault = hasDefault
+
+/**
+ * @param {Reducer} reducer
+ * @throws if reducer is not valid
+ * @return {Function}
+ */
+function getResolveFunction (reducer) {
+  const reducerType = reducers[reducer.type]
+  if (reducerType) {
+    return reducerType.resolve
+  }
+
+  throw new Error(`Reducer type '${reducer.type}' was not recognized`)
+}
+
 /**
  * apply a Reducer to an accumulator
  * @param {Object} manager
  * @param {Accumulator} accumulator
  * @param {Reducer} reducer
  * @returns {Promise<Accumulator>}
  */
-function resolve (manager, accumulator, reducer) {
+function resolveReducer (manager, accumulator, reducer) {
   // this conditional is here because BaseEntity#resolve
   // does not check that lifecycle methods are defined
   // before trying to resolve them
   if (!reducer) {
     return Promise.resolve(accumulator)
   }
 
-  const reducerType = reducers[reducer.type]
-  if (!reducerType) {
-    throw new Error(`Reducer type '${reducer.type}' was not recognized`)
+  let resolve = getResolveFunction(reducer)
```

Comment:
why are you overwriting resolve here? above line is also setting it, any special reason?

Reply:
overwriting it just because it's one less variable name to think up